### PR TITLE
Example application

### LINKED ISSUE #119: Create ExampleApplication component
Now this is a monorepo I think it may make sense to have `packages/example` being the govuk-react-example project.

Lerna lets you import another repo keeping commit history, we should probably import using this.

### LINKED ISSUE #815: Set up Netlify with OSS team
We have a Netlify OSS sponsored Team, I need to set this up.

### PR DESCRIPTION
**Pull request by penx**
An example application for the purpose of:

- Reviewing API in situ
- Visual regression testing
- Integration testing
- Measuring code coverage of integration tests

When it comes to visual regression, the plan would be for different pages or application states to be preloaded in to their own story.

Closes #815
Fixes #119

### CODE REVIEW

File: packages/example-application/src/form.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    if (!firstName) {
+      setErrors({ firstName: 'Please enter a first name' });
+    } else {
+      setIsSubmitting(true);
+      setTimeout(() => {
+        setIsSubmitted(true);
+        setIsSubmitting(false);
+      }, 1000);
+    }
+  });
```
Comment (stevesims) [inline packages/example-application/src/form.js:21]: hmm - eslint should have been whining at you about the missing dependencies list for this function :grin:

**Review by stevesims — APPROVED** [review]
LGTM

couple of minor comments :grin:

File: packages/example-application/src/index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import * as GovUK from 'govuk-react';
+import { MemoryRouter as Router, Switch, Route } from 'react-router';
+
+import { createGlobalStyle } from 'styled-components';
+import Home from './home';
+import Form from './form';
+
+const GlobalStyle = createGlobalStyle`
```
[the remark this answers is not in the document]
Comment (penx) [reply packages/example-application/src/index.js:9]: am planning this :-) 

and yes, the purpose of the ExampleApplication is to review the API we have and figure out what we don't like or what is missing

File: packages/example-application/src/form.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import React, { useState, useCallback } from 'react';
+import * as GovUK from 'govuk-react';
```
[the remark this answers is not in the document]
Comment (penx) [reply packages/example-application/src/form.js:2]: I want to import from the govuk package and not sub packages in order to prove everthing that's needed to get 100% code coverage can be accessed via the main export.

Yes perhaps I shouldn't do `* as` considering our docs don't state this, but I quite like it as I don't have to think about imports every time I decide to write a new element - makes it much faster.

File: packages/example-application/src/form.js (hunk 1 above)
Comment (penx) [reply packages/example-application/src/form.js:21]: https://github.com/govuk-react/govuk-react/issues/827

File: packages/example-application/src/form.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (penx) [reply packages/example-application/src/form.js:2]: https://github.com/govuk-react/govuk-react/issues/424#issuecomment-844407897

File: packages/example-application/src/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (penx) [reply packages/example-application/src/index.js:9]: https://github.com/govuk-react/govuk-react/issues/424#issuecomment-844407897

[the remark this answers is not in the document]
Comment (stevesims) [reply packages/example-application/src/index.js:9]: :100:

**Pull request merged**